silences-gitops/README.md

```diff
@@ -0,0 +1,114 @@
+
+This solution is laid out as follows:
+
+- Silence CRs common to all clusters are stored in the MCB repository, under the `bases/silences/` directory
+- Silence CRs added in MCB also need to be referenced in the `bases/silences/kustomization.yaml` file
+- Silence CRs specific to a cluster are stored in the corresponding CMC repository, under the `management-clusters/<management-cluster-name>/silences/` directory
+- Name collision between MCB and CMC Silences CRs is prevented by adding the special `global-` name prefix when deploying Silences CRs from the MCB repository
+
+![Silences GitOps Architecture](silences-gitops.png)
+
+### Examples
+
+- Common Silence CR in MCB repository:
+
+```yaml
+$ cat management-cluster-bases/bases/silences/alice.yaml
+apiVersion: monitoring.giantswarm.io/v1alpha1
+kind: Silence
+metadata:
+  name: alice
+  annotations:
+    valid-until: "2100-01-01"
+spec:
+  matchers:
+  - name: alertname
+    value: AliceAlert
+    isRegex: false
+```
+
+```yaml
+$ cat management-cluster-bases/bases/silences/kustomization.yaml
+resources:
+- alice.yaml
+```
+
+- Cluster specific Silence CR in CMC repository:
+
+```yaml
+$ cat giantswarm-management-clusters/management-clusters/some-management-cluster/silences/bob.yaml
+apiVersion: monitoring.giantswarm.io/v1alpha1
+kind: Silence
+metadata:
+  name: bob
+  annotations:
+    valid-until: "2100-01-01"
+spec:
+  matchers:
+  - name: alertname
+    value: BobAlert
+    isRegex: false
+  - name: cluster_id
+    value: some-cluster-id
+    isRegex: false
+```
+
+### Use cases
+
+- Create a silence for all clusters
+  - Add the Silence CR to MCB repository in `bases/silences/` directory
+  - Reference the new Silence CR in the `bases/silences/kustomization.yaml` file
+
+- Create a silence for a single cluster
+  - Add the Silence CR to the `management-clusters/some-management-cluster/silences/` directory
+  - Add a matcher for the `cluster_id` tag with the cluster ID as value in the Silence CR
+
+- Create a silence for multiple clusters in a single management cluster
+  - Add the Silence CR to the `management-clusters/some-management-cluster/silences/` directory
+  - (Optionally) Add a matcher for the `cluster_id` tag with all cluster ID as regex value in the Silence CR
+
+- Create a silence for multiple clusters in a multiple management clusters
+  - Add the Silence CR for each desired management cluster in `management-clusters/<each-management-cluster>/silences/` directory
+  - (Optionally) Add a matcher for the `cluster_id` tag in the Silence CR
+
+### Benefits
+
+- Silences are deployed and managed using standard GitOps mechanisms
+- Customers can manage their own silences
+- Fixes current privacy and security issues
+
+### Drawbacks
+
+- Rather than using the `targetTags` field to define where the Silence CR is deployed, user now needs to duplicate the Silence CR for each management cluster it needs to be deployed to
```

Comment:
How is that a drawback? That feels like an advantage to me, no more messing up with regex and so on

Reply:
I may be wrong but if a silence needs to be in multiple customers then it should be a global silence anyway imo

Reply:
I meant that I do not see a use case where we would have the same silence for 2 different customers that should not be deployed for all installations

Reply:
It is more likely than it should be a provider, region or kaas release specific alert and we should strive to find a better common denomiator than the customer name (which should only be added when pushing data to grafana cloud anyway)

Reply:
Proposed something for providers above. Also if a silence needs to be applied to multiple MC of a single customer, it just a matter of code organization: add a `bases/silences` folder in the CMC repo and reference the silences in the MC specific code.

Reply:
What's your proposition for providers ?

Reply:
Taking this into consideration for the implementation phase